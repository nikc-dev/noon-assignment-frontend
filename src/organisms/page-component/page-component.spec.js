import React from 'react';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from 'enzyme-to-json';
import Page from './index';
configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));
it('renders correctly', () => {
  const component = renderer.create(
    <Page />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correct page based on toggle value', () => {
  const component = mount(<Page />);
 
  component.find('#navbar-liked').simulate('click')
  expect(toJson(component)).toMatchSnapshot();
});