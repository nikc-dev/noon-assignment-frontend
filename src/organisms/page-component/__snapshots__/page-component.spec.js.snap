// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct page based on toggle value 1`] = `
<Page>
  <div
    className="page"
  >
    <NavBar
      isMobile={false}
      toggle={[Function]}
    >
      <div>
        <nav
          className="navbar"
        >
          <h1>
             :) noon assignment
          </h1>
          <div
            className="navbar__links"
          >
            <a
              href="/"
              id="navbar-home"
              onClick={[Function]}
            >
              Home
            </a>
            <a
              href="/"
              id="navbar-liked"
              onClick={[Function]}
            >
              Liked
            </a>
          </div>
        </nav>
      </div>
    </NavBar>
    <Liked>
      <div
        className="liked"
      >
        <Loader>
          <div
            className="loader"
          />
        </Loader>
      </div>
    </Liked>
  </div>
</Page>
`;

exports[`renders correctly 1`] = `
<div
  className="page"
>
  <div>
    <nav
      className="navbar"
    >
      <h1>
         :) noon assignment
      </h1>
      <div
        className="navbar__links"
      >
        <a
          href="/"
          id="navbar-home"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          href="/"
          id="navbar-liked"
          onClick={[Function]}
        >
          Liked
        </a>
      </div>
    </nav>
  </div>
  <div
    className="home"
  >
    <div
      className="loader"
    />
  </div>
</div>
`;
